lib/icingadb/icingadb-objects.cpp

```diff
@@ -1383,8 +1383,11 @@ bool IcingaDB::PrepareObject(const ConfigObject::Ptr& object, Dictionary::Ptr& a
 			attributes->Set("timeperiod_id", GetObjectIdentifier(timeperiod));
 
 		if (notification->GetTimes()) {
-			attributes->Set("times_begin", notification->GetTimes()->Get("begin"));
-			attributes->Set("times_end",notification->GetTimes()->Get("end"));
+			auto begin (notification->GetTimes()->Get("begin"));
+			auto end (notification->GetTimes()->Get("end"));
+
+			attributes->Set("times_begin", begin == Empty || (double)begin < 0 ? Empty : Value(std::round((double)begin)));
+			attributes->Set("times_end", end == Empty || (double)end < 0 ? Empty : Value(std::round((double)end)));
```

Comment:
I think a simple if would be easier to read here:

```c++
if (begin != Empty && (double)begin >= 0) {
    attributes->Set("times_begin", Value(std::round((double)begin)));
}
```

I also don't see another advantage in a ternary operator here. `times_{begin,end}` is only added conditionally to the JSON object anyways.